src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -746,6 +748,15 @@ class WindowSpecificationSegment(BaseSegment):
     )
 
 
+@ansi_dialect.segment()
+class FunctionNameSegment(BaseSegment):
```

Comment:
This new segment essentially replaces the old regex-based `FunctionNameSegment` so that rules looking for `function_name` in the parse tree are guaranteed to find it regardless of quoting.